openapi-core/src/main/java/io/ballerina/openapi/core/generators/common/OASModifier.java

```diff
@@ -116,137 +116,59 @@ public OpenAPI modifyWithBallerinaConventions(OpenAPI openapi, Map<String, Strin
         return openapi;
     }
 
-    private static void modifyOASWithSchemaName(OpenAPI openapi, Map<String, String> nameMap) {
-        Components components = openapi.getComponents();
+    public OpenAPI modifyWithBallerinaConventions(OpenAPI openapi) {
+        Optional<Map<String, String>> proposedNameMap = getProposedNameList(openapi);
```

Comment:
Any reason to return an `Map` wrapped with `Optional` here ? IMO if we have an empty collection, we should use either `Collections.emptyList` or `Collections.emptyMap` in those case (depending on the requirement).

Reply:
Fixed with [11693ad](https://github.com/ballerina-platform/openapi-tools/pull/1766/commits/11693ad44baf485c02e35d48419e727fba710c02)